src/components/Board.tsx

```diff
@@ -1,77 +1,85 @@
 import { useAnimationFrame } from "@hooks/useAnimationFrame";
+import { useGameContext } from "@/hooks/useGameContext";
 import { useKeyboard } from "@hooks/useKeyboard";
 import Grid from "./Grid";
 import Snake from "./Snake";
 import styles from "./Board.module.scss";
+import Food from "./Food";
 
 const Board: FC = (): React.ReactElement => {
-  const [snake, setSnake] = useState(() => SnakeSprite.Create(BOARD_SIZE));
-  const [velocity, setVelocity] = useState({ x: 1, y: 0 });
+  const {
+    isMoving,
+    direction,
+    changeMovement,
+    start,
+    pause,
+    animate,
+    snake,
+    food,
+  } = useGameContext();
 
-  const animate = useCallback(() => {
-    setSnake((prev) => {
-      const current = new SnakeSprite([...prev.segments]);
-      current.move({
-        x: velocity.x,
-        y: velocity.y,
-      });
-
-      return current;
-    });
-  }, [velocity]);
-
-  const keypressHandler = (event: KeyboardEvent) => {
-    const keyPressed = event.code;
-
-    const movingUp = velocity.y === -1;
-    const movingDown = velocity.y === 1;
-    const movingLeft = velocity.x === -1;
-    const movingRight = velocity.x === 1;
+  const handleKeyPressed = (keyCode: string) => {
+    const movingUp = direction.y === -1;
+    const movingDown = direction.y === 1;
+    const movingLeft = direction.x === -1;
+    const movingRight = direction.x === 1;
 
     switch (true) {
-      case keyPressed === "ArrowUp" && !movingDown:
-        setVelocity({
+      case keyCode === "ArrowUp" && !movingDown:
+        changeMovement({
           x: 0,
           y: -1,
         });
         break;
-      case keyPressed === "ArrowDown" && !movingUp:
-        setVelocity({
+      case keyCode === "ArrowDown" && !movingUp:
+        changeMovement({
           x: 0,
           y: 1,
         });
         break;
-      case keyPressed === "ArrowLeft" && !movingRight:
-        setVelocity({
+      case keyCode === "ArrowLeft" && !movingRight:
+        changeMovement({
           x: -1,
           y: 0,
         });
         break;
-      case keyPressed === "ArrowRight" && !movingLeft:
-        setVelocity({
+      case keyCode === "ArrowRight" && !movingLeft:
+        changeMovement({
           x: 1,
           y: 0,
         });
         break;
+      case keyCode === "Enter":
+        start();
+        break;
+      case keyCode === "Space":
+        pause();
+        break;
     }
   };
 
-  useKeyboard(keypressHandler);
+  useKeyboard(handleKeyPressed);
   useAnimationFrame({
     nextAnimationFrameHandler: animate,
-    framePerSeconds: SNAKE_SPEED / 4,
+    isActive: isMoving,
   });
 
+  const boadStyle = {
+    gridTemplateRows: `repeat(${BOARD_SIZE}, ${BORDER_UNIT_SIZE}px)`,
+    gridTemplateColumns: `repeat(${BOARD_SIZE}, ${BORDER_UNIT_SIZE}px)`,
+  };
+
   return (
-    <div className={styles.board}>
-      <Grid size={BOARD_SIZE} />
+    <div className={styles.board} style={boadStyle}>
```

Comment:
So technically you can use inline styles. But preferably stick to one styling method. (there are exceptions though). Say can we consider dynamically pass classname and apply styling based on it?